test/e2e/test-errortracker.js

```diff
@@ -104,303 +116,311 @@ describe('Error Tracker Server', () => {
     sandbox.restore();
   });
 
-  describe('rejects bad requests', () => {
-    it('without referrer', () => {
-      return makeRequest('', knownGoodQuery).then(() => {
-        throw new Error('UNREACHABLE');
-      }, (err) => {
-        expect(err.message).to.equal('Bad Request');
-      });
-    });
+  method("GET", makeRequest);
+  method("POST JSON", makePostRequest('json'));
+  method("POST Text", makePostRequest('text/plain'));
+
+  function method(description, makeRequest) {
```

Comment:
I'm sure we can think of a better name for this. `requestTestSuite()`?

Reply:
Done.